spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/ParquetWithSparkSchemaVisitor.java

```diff
@@ -59,106 +59,101 @@ public static <T> T visit(DataType sType, Type type, ParquetWithSparkSchemaVisit
     } else {
       // if not a primitive, the typeId must be a group
       GroupType group = type.asGroupType();
-      OriginalType annotation = group.getOriginalType();
+      LogicalTypeAnnotation annotation = group.getLogicalTypeAnnotation();
       if (annotation != null) {
-        switch (annotation) {
-          case LIST:
-            Preconditions.checkArgument(
-                !group.isRepetition(Repetition.REPEATED),
-                "Invalid list: top-level group is repeated: %s",
-                group);
-            Preconditions.checkArgument(
-                group.getFieldCount() == 1,
-                "Invalid list: does not contain single repeated field: %s",
-                group);
-
-            GroupType repeatedElement = group.getFields().get(0).asGroupType();
-            Preconditions.checkArgument(
-                repeatedElement.isRepetition(Repetition.REPEATED),
-                "Invalid list: inner group is not repeated");
-            Preconditions.checkArgument(
-                repeatedElement.getFieldCount() <= 1,
-                "Invalid list: repeated group is not a single field: %s",
-                group);
-
-            Preconditions.checkArgument(
-                sType instanceof ArrayType, "Invalid list: %s is not an array", sType);
-            ArrayType array = (ArrayType) sType;
-            StructField element =
-                new StructField(
-                    "element", array.elementType(), array.containsNull(), Metadata.empty());
-
-            visitor.fieldNames.push(repeatedElement.getName());
-            try {
-              T elementResult = null;
-              if (repeatedElement.getFieldCount() > 0) {
-                elementResult = visitField(element, repeatedElement.getType(0), visitor);
-              }
-
-              return visitor.list(array, group, elementResult);
-
-            } finally {
-              visitor.fieldNames.pop();
+        if (annotation.equals(LogicalTypeAnnotation.listType())) {
```

Comment:
I would recommend reviewing this changer with "Hide whitespace" option. 
I changed `switch` to `if` because `LogicalTypeAnnotation` isn't enum.